src/odemis/util/img.py

```diff
@@ -811,3 +813,105 @@ def getBoundingBox(content):
 
     return rect
 
+
+class ImageIntegrator(object):
+    """
+    Integrate the images one after another. Once the first image is acquired, calculate the best type for fitting
+    the image to avoid saturation and overflow. At the end of acquisition, take the average of integrated data if
+    the detector is DT_NORMAL and subtract the baseline from the final integrated image.
+    """
+    def __init__(self, steps):
+        """
+        steps: (int) the total number of images that need to be integrated
+        """
+        self.steps = steps  # can be changed by the caller, on the fly
+        self._step = 0
+        self._img = None
+        self._best_dtype = None
+
+    def append(self, img):
+        """
+        Integrate two images (the new acquired image with the previous integrated one if exists) and return the
+        new integrated image. It will reset the ._img after reaching the number of integration counts, notifying
+        that the integration of the acquired images is completed.
+        Returns:
+            model.DataArray: the integrated image with the updated metadata
+        """
+        self._step += 1
+
+        if self._img is None:
+            orig_dtype = img[-1].dtype
+            self._best_dtype = get_best_dtype_for_acc(orig_dtype, self.steps)
+            self._img = img
+            return img
+        else:
+            # get the metadata from the last acquired image (should be the same for all images)
+            md = img[-1].metadata
+            logging.debug("Integrate one image after another")
+            data = numpy.add(img[0], img[1], dtype=self._best_dtype)
+            # update the metadata of the integrated image in every integration step
+            md = self.update_integration_metadada(md)
+            self._img = [model.DataArray(data, md)]
+
+            # At the end of the acquisition, check if the detector type is DT_NORMAL and then take the average by
+            # dividing with the number of acquired images (integration count) for every pixel position and restoring
+            # the original dtype.
+            if self._step == self.steps:
```

Comment:
Make it `>=`, so that if the user changed steps to a smaller number, it doesn't go forever.